diff --git a/cmake/vcpkg-ports/cpuinfo/portfile.cmake b/cmake/vcpkg-ports/cpuinfo/portfile.cmake
index 67bd18e..9c8610d 100644
--- a/cmake/vcpkg-ports/cpuinfo/portfile.cmake
+++ b/cmake/vcpkg-ports/cpuinfo/portfile.cmake
@@ -11,6 +11,7 @@ vcpkg_from_github(
     HEAD_REF master
     PATCHES
         patch_cpuinfo_h_for_arm64ec.patch
+        patch_cpuinfo_h_for_loong64.patch
         patch_vcpkg_arm64ec_support.patch       # https://github.com/pytorch/cpuinfo/pull/324
         win_arm_fp16_detection_fallback.patch   # https://github.com/pytorch/cpuinfo/pull/348
 )
diff --git a/onnxruntime/core/mlas/lib/sqnbitgemm_kernel_lasx.cpp b/onnxruntime/core/mlas/lib/sqnbitgemm_kernel_lasx.cpp
index a43043e..c782fb8 100644
--- a/onnxruntime/core/mlas/lib/sqnbitgemm_kernel_lasx.cpp
+++ b/onnxruntime/core/mlas/lib/sqnbitgemm_kernel_lasx.cpp
@@ -32,7 +32,8 @@ QNBitGemmPackQuantBDataSize_Lasx(
     size_t K,
     size_t BlkLen,
     bool /* HasZeroPoint */,
-    MLAS_QNBIT_GEMM_COMPUTE_TYPE ComputeType
+    MLAS_QNBIT_GEMM_COMPUTE_TYPE ComputeType,
+    const MLAS_BACKEND_KERNEL_SELECTOR_CONFIG*
 )
 {
     const size_t BlockCountK = MlasDivRoundup(K, BlkLen);
diff --git a/tools/ci_build/github/linux/build_linux_python_package.sh b/tools/ci_build/github/linux/build_linux_python_package.sh
index 27bf6f9..76147e4 100755
--- a/tools/ci_build/github/linux/build_linux_python_package.sh
+++ b/tools/ci_build/github/linux/build_linux_python_package.sh
@@ -113,7 +113,7 @@ do
   ls "$python3_dir"
   ${PYTHON_EXE} -m pip install -r /onnxruntime_src/tools/ci_build/github/linux/python/requirements.txt
   PATH=$python3_dir:$PATH ${PYTHON_EXE} /onnxruntime_src/tools/ci_build/build.py "${BUILD_ARGS[@]}"
-  cp /build/"$BUILD_CONFIG"/dist/*.whl /build/dist
+  auditwheel repair -w wheelhouse /build/"$BUILD_CONFIG"/dist/*.whl
 done
 
 if command -v ccache &> /dev/null; then
